src/base/utils/string.h

```diff
@@ -37,18 +37,37 @@
 #include <Qt>
 #include <QtContainerFwd>
 
+#ifndef Q_OS_WIN
+#include <QCollator>
+#endif
+
 class QStringRef;
 
 namespace Utils::String
 {
-    QString fromDouble(double n, int precision);
-
-    int naturalCompare(const QString &left, const QString &right, const Qt::CaseSensitivity caseSensitivity);
-    template <Qt::CaseSensitivity caseSensitivity>
-    bool naturalLessThan(const QString &left, const QString &right)
+    class NaturalCompare
     {
-        return (naturalCompare(left, right, caseSensitivity) < 0);
-    }
+    public:
+        explicit NaturalCompare(const Qt::CaseSensitivity caseSensitivity = Qt::CaseSensitive);
+        int operator()(const QString &left, const QString &right) const;
+
+    private:
+#ifdef Q_OS_WIN
+        const Qt::CaseSensitivity m_caseSensitivity;
+#else
+        QCollator m_collator;
+#endif
+    };
+
+    class NaturalCompareLess
```

Comment:
Why not `NaturalLessThan` (as it was before)? IMO, having both Compare and Less together is a bit overkill.

Reply:
We still could have just stub-functions here and put classes in .cpp.

Reply:
>Why not NaturalLessThan (as it was before)? IMO, having both Compare and Less together is a bit overkill.

Renamed.

>We still could have just stub-functions here and put classes in .cpp.

That is not possible if I want to use something like `std::set<QString, NaturalLessThan<Qt::CaseSensitive>>`.

Reply:
>That is not possible if I want to use something like `std::set<QString, NaturalLessThan<Qt::CaseSensitive>>`.

Ok. Nevermind.